lib/jwt/jwk/key_base.rb

```diff
@@ -50,6 +52,83 @@ def <=>(other)
       private
 
       attr_reader :parameters
+
+      KEY_USAGES_SIG  = ['Digital Signature', 'Non Repudiation', 'Content Commitment', 'Key Cert Sign', 'CRL Sign'].freeze
```

Comment:
Im wondering about where this certificate specific stuff should live. Now the logic baked into each and every JWK object created. 

Most of these methods are just setting some values under certain keys? Could it just be a separate mechanism (method/extension) that would build the params passed into the key or add the params to an existing key?